src/Prelude/Linear/Internal/Simple.hs

```diff
@@ -56,3 +56,6 @@ foldr :: (a ->. b ->. b) -> b ->. [a] ->. b
 foldr f z = \case
   [] -> z
   x:xs -> f x (foldr f z xs)
+
+lazyPair :: (x ->. y ->. z) ->. (x,y) ->. z
+lazyPair k = Unsafe.toLinear $ \v -> k (Prelude.fst v) (Prelude.snd v)
```

Comment:
That looks dangerous. Let's just drop lazy state from our list of supported instances and avoid using `Unsafe.toLinear` gratuitously.

Reply:
Hmm, okay. My hope was that since both components of the pair are consumed, this should indeed be linear.

Reply:
Well, it may be considered linear. But it is very subtle: it is because of call-by-need/laziness that we can say that the spine of the pair is forced exactly once, so it's ok, somehow to traverse it twice (remember: there may be linear things in a thunk). Under a call-by-name semantics this is clearly not permitted.

Now, GHC _probably_ never unshares thunk, well, at least not if there is some real computation in them, _probably_. But it might. Which would probably make some things slower in non-linear code, but could possibly segfault in linear code. Do we want to take this risk?

The bottom line is that there is no plan, at the moment, to teach the type system (in particular the linter) about the interaction of sharing with linearity. And in this perspective, it's way to bold to add this `lazyPair` primitive.

Reply:
Fair enough! Is there a plan to somehow allow lazy consumption of a pair in this sort of way?

Reply:
Not at the moment. I don't even know how to approach it.